src/Map/src/Twig/MapRuntime.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\Map\Twig;
+
+use Symfony\UX\Map\Map;
+use Symfony\UX\Map\Marker;
+use Symfony\UX\Map\Point;
+use Symfony\UX\Map\Renderer\RendererInterface;
+use Twig\Extension\RuntimeExtensionInterface;
+
+/**
+ * @author Simon André <smn.andre@gmail.com>
+ *
+ * @internal
+ */
+final class MapRuntime implements RuntimeExtensionInterface
+{
+    public function __construct(
+        private readonly RendererInterface $renderer,
+    ) {
+    }
+
+    /**
+     * @param array<string, mixed> $attributes
+     */
+    public function renderMap(
+        ?Map $map = null,
+        array $attributes = [],
+        ?array $center = null,
+        ?float $zoom = null,
+        ?array $markers = null,
+    ): string {
+        if (!$map instanceof Map) {
+            $map = new Map();
+        }
+        if (null !== $center) {
+            $map->center(Point::fromArray($center));
+        }
+        if (null !== $zoom) {
+            $map->zoom($zoom);
+        }
+        if (null !== $markers) {
+            foreach ($markers as $marker) {
+                $map->addMarker(Marker::fromArray($marker));
+            }
+        }
```

Comment:
If the user pass a `Map` and `center` option, it will actually mutate the `Map` instance, are we sure about that? 

Shouldn't we call `->center`, `->zoom`, `->addMarker` only when the user didn't pass a `Map`?

Reply:
I was wondering too. Maybe we should do some cloning here ?

Reply:
We can, but I don't know how much it can impact performances. 

I think we can throw an error if the user try to re-configure center/zoom/markers on-the-fly, on an already existing Map. 

WDYT?

Reply:
Better now ?

Reply:
Yes, thanks!